streams/src/main/java/org/apache/kafka/streams/state/internals/InMemoryWindowStore.java

```diff
@@ -406,16 +434,26 @@ public Long peekNextKey() {
         }
     }
 
-    private class WrappedWindowedKeyValueIterator extends InMemoryWindowStoreIteratorWrapper implements KeyValueIterator<Windowed<Bytes>, byte[]> {
+    private static class WrappedWindowedKeyValueIterator extends InMemoryWindowStoreIteratorWrapper implements KeyValueIterator<Windowed<Bytes>, byte[]> {
+
+        private final boolean retainDuplicates;
+        private final long windowSize;
 
         WrappedWindowedKeyValueIterator() {
             super();
+            this.windowSize = 0;
+            this.retainDuplicates = false;
```

Comment:
You actually do not need `this` here, right? 

The values are the default initialization values in Java. And `super` is also called in the default constructor. So actually, you could remove this constructor completely.

Reply:
I did seem to need this constructor here but I'm removing all the dummy constructors in favor of an emptyIterator() supplier